arangod/Cache/TransactionManager.cpp

```diff
@@ -33,81 +33,68 @@
 namespace arangodb::cache {
 
 TransactionManager::TransactionManager()
-    : _openReads(0), _openSensitive(0), _openWrites(0), _term(0), _lock(false) {}
+    : _data({{0,0,0}, 0}) {}
 
 Transaction* TransactionManager::begin(bool readOnly) {
   Transaction* tx = new Transaction(readOnly);
 
-  lock();
-
+  Data newData;
   if (readOnly) {
-    _openReads++;
-    if (_openWrites.load() > 0) {
-      tx->sensitive = true;
-      _openSensitive++;
-    }
+    Data data = _data.load(std::memory_order_relaxed);
+    do {
+      tx->sensitive = false;
+      newData = data;
+      newData.counters.openReads++;
+      if (newData.counters.openWrites > 0) {
+        tx->sensitive = true;
+        newData.counters.openSensitive++;
+      }
+    } while (!_data.compare_exchange_strong(data, newData, std::memory_order_acq_rel, std::memory_order_relaxed));
   } else {
     tx->sensitive = true;
-    if (_openSensitive.load() == 0) {
-      _term++;
-    }
-    if (_openWrites.load() == 0) {
-      _openSensitive = _openReads.load() + _openWrites.load();
-    }
-    _openWrites++;
-    _openSensitive++;
+    Data data = _data.load(std::memory_order_relaxed);
+    do {
+      newData = data;
+      newData.counters.openWrites++;
+      if (newData.counters.openSensitive == 0) {
+        newData.term++;
+      }
+      if (newData.counters.openWrites++ == 0) {
+        newData.counters.openSensitive = data.counters.openReads + 1;
+      } else {
+        newData.counters.openSensitive++;
+      }
+    } while (!_data.compare_exchange_strong(data, newData, std::memory_order_acq_rel, std::memory_order_relaxed));
   }
-
-  tx->term = _term.load();
-
-  unlock();
-
+  tx->term = newData.term;
   return tx;
 }
 
 void TransactionManager::end(Transaction* tx) noexcept {
   TRI_ASSERT(tx != nullptr);
-  lock();
-
-  // if currently in sensitive phase, and transaction term is old, it was
-  // upgraded to sensitive status
-  if (((_term & static_cast<uint64_t>(1)) > 0) && (_term > tx->term)) {
-    tx->sensitive = true;
-  }
 
-  if (tx->readOnly) {
-    _openReads--;
-  } else {
-    _openWrites--;
-  }
+  Data data = _data.load(std::memory_order_relaxed);
+  Data newData;
+  do {
+    if (((data.term & static_cast<uint64_t>(1)) > 0) && (data.term > tx->term)) {
+      tx->sensitive = true;
+    }
 
-  if (tx->sensitive && (--_openSensitive == 0)) {
-    _term++;
-  }
+    auto newData = data;
```

Comment:
```suggestion
    newData = data;
```

Reply:
Our CI caught this one, led to an uninitialized value use.

Reply:
Ah, I missed that in the last refactoring step. This also explains the test failures... unittests ran successfully now on my machine.